pkg/services/publicdashboards/service/service.go

```diff
@@ -332,7 +332,19 @@ func (pd *PublicDashboardServiceImpl) GetOrgIdByAccessToken(ctx context.Context,
 	return pd.store.GetOrgIdByAccessToken(ctx, accessToken)
 }
 
-func (pd *PublicDashboardServiceImpl) Delete(ctx context.Context, uid string) error {
+func (pd *PublicDashboardServiceImpl) Delete(ctx context.Context, uid string, dashboardUid string) error {
+	// get existing public dashboard if exists
+	existingPubdash, err := pd.store.Find(ctx, uid)
+	if err != nil {
+		return ErrInternalServerError.Errorf("Delete: failed to find public dashboard by uid: %s: %w", uid, err)
+	} else if existingPubdash == nil {
+		return ErrPublicDashboardNotFound.Errorf("Delete: public dashboard not found by uid: %s", uid)
+	}
```

Comment:
```suggestion
	if err != nil {
		return ErrInternalServerError.Errorf("Delete: failed to find public dashboard by uid: %s: %w", uid, err)
	}
	
	if existingPubdash == nil {
		return ErrPublicDashboardNotFound.Errorf("Delete: public dashboard not found by uid: %s", uid)
	}
```
nit: since you are returning after each if, you could change this code to separate error handling and also avoid nested ifs

Reply:
is there an existing an error-handling file? or do you suggest I create a new one